server/src/main/java/io/crate/planner/statement/DeletePlanner.java

```diff
@@ -193,4 +203,32 @@ private static ExecutionPlan deleteByQuery(DocTableRelation table, PlannerContex
         Collect collect = new Collect(collectPhase, TopN.NO_LIMIT, 0, 1, 1, null);
         return Merge.ensureOnHandler(collect, context, Collections.singletonList(MergeCountProjection.INSTANCE));
     }
+
+    private static class NonPartitionReferenceDetector extends SymbolVisitor<List<Reference>, Boolean> {
```

Comment:
Would it be possible to use the existing `SymbolVisitors.any` or `Symbols.containsColumn`  for this?